### PR TITLE
Feature/qs menu

### PR DESCRIPTION
**Pull request by kylelengling**
Quick start menu. Basic functionality and styling to match the sitecore styles. Currently absolutely positioned because it's only being used in the navigation. Left a note to revisit if we ever use it outside of the navigation.

### CODE REVIEW

File: components/QuickStartMenu.tsx [hunk 1]
```diff
@@ -24,6 +67,62 @@ const QuickStartMenu = ({ className }: QuickStartMenuProps): JSX.Element => {
...
+                href={child.url}
+                target="_blank"
+              >
+                {child.title}
+                <svg
+                  width="16"
+                  height="16"
+                  xmlns="http://www.w3.org/2000/svg"
+                  className={classnames('transform-gpu', 'scale-625')}
+                >
```
Comment (andershopkins) [inline components/QuickStartMenu.tsx:114]: We can avoid having to add another sizing variable for `scale` by doing something like this instead:

```
<svg
  xmlns="http://www.w3.org/2000/svg"
  viewBox="0 0 16 16"
  className={classnames('w-em', 'h-em', 'scale-75', 'transform-gpu')}
```

By specifying `viewBox` we actually scale the icon based on `em` sizing, then pop the `scale-75` default TW class on there and boom, good to go.

File: components/QuickStartMenu.tsx [hunk 2]
```diff
@@ -24,6 +67,62 @@ const QuickStartMenu = ({ className }: QuickStartMenuProps): JSX.Element => {
...
+          'bg-white',
+          'border',
+          'border-gray-light',
+          'absolute',
+          'top-20',
+          'w-56',
+          'right-1',
+          'lg:transform-none',
+          'lg:transition-none',
+          'transform-gpu',
```
Comment (andershopkins) [inline components/QuickStartMenu.tsx:84]: Are these `transform` and `transition` utilities being actually used for anything here? If the element is going from `display: none` to `display:whatever` - what's the purpose?

File: components/QuickStartMenu.tsx [hunk 3]
```diff
@@ -1,15 +1,58 @@
...
+      window.removeEventListener('click', pageClickEvent);
+    };
+  }, [isOpen]);
+
+  const quickStartMenuInactiveClasses = classnames(
+    'translate-x-full',
+    'lg:translate-none',
+    'lg:hidden'
+  );
+  const quickStartMenuActiveClasses = classnames('translate-x-0');
```
Comment (andershopkins) [inline components/QuickStartMenu.tsx:48]: Same here. If we remove all the `translate` properties, won't the affect be exactly the same?

File: components/QuickStartMenu.tsx [hunk 4]
```diff
@@ -24,6 +67,62 @@ const QuickStartMenu = ({ className }: QuickStartMenuProps): JSX.Element => {
...
+          'lg:transform-none',
+          'lg:transition-none',
+          'transform-gpu',
+          {
+            [quickStartMenuInactiveClasses]: !isOpen,
+            [quickStartMenuActiveClasses]: isOpen,
+          }
+        )}
+      >
+        <div className={classnames('hidden')}>{SitecoreQuickLinks.title}</div>
```
Comment (andershopkins) [inline components/QuickStartMenu.tsx:91]: Two things:

1. `{SitecoreQuickLinks.title}` should probably `sr-only` text inside of the `<button>` itself.
2. Remember that `hidden` is `display: none` - use the `sr-only` class when appropriate.

File: components/QuickStartMenu.tsx [hunk 5]
```diff
@@ -24,6 +67,62 @@ const QuickStartMenu = ({ className }: QuickStartMenuProps): JSX.Element => {
           </g>
         </svg>
       </button>
+      {/* NOTE:: Dropdown currently absolutely positioned. Revist if used in place other than the navigation */}
+      <div
```
Comment (andershopkins) [inline components/QuickStartMenu.tsx:71]: You'll want to add `id` elements to the `button` and this wrapping `<div>`, use `react-id-generator` - you can find an example in `SearchBox.tsx`. 

Then you'll also want `aria-labelledby` on this element, pointing to the `id` of the `<button>` element.

File: components/QuickStartMenu.tsx [hunk 6]
```diff
@@ -1,15 +1,58 @@
...
+  );
+  const quickStartMenuActiveClasses = classnames('translate-x-0');
+
   return (
     <div className={classnames('block', 'w-5', 'h-5', className)}>
-      <button className={classnames('w-inherit', 'h-inherit', 'hover:text-teal', 'transition')}>
+      <button
+        className={classnames('w-inherit', 'h-inherit', 'hover:text-teal', 'transition')}
+        onClick={toggleQuickStartMenu}
+      >
```
Comment (andershopkins) [inline components/QuickStartMenu.tsx:55]: Once you have and `id` and the `sr-only` title inside of this `<button>`, you can also add `aria-expanded` and have that toggle from `false` to `true` with `isOpen`.

**Review by andershopkins — CHANGES_REQUESTED** [review]

File: components/QuickStartMenu.tsx [hunk 7]
```diff
@@ -1,15 +1,58 @@
...
 
-const QuickStartMenu = ({ className }: QuickStartMenuProps): JSX.Element => {
+const QuickStartMenu = ({ className, callback }: QuickStartMenuProps): JSX.Element => {
+  const quickStartMenuRef = useRef<HTMLDivElement>(null);
+  const [isOpen, setOpen] = useState(false);
+  const toggleQuickStartMenu = (event: React.MouseEvent) => {
+    if (callback && (event?.target as HTMLButtonElement)?.localName !== 'button') {
+      console.log(event.target);
+      callback();
+    }
```
Comment (bhgeorge) [inline components/QuickStartMenu.tsx:20]: Is there any reason for the callback here? It seems like this button is pretty internal to the component. Curious if we have a use case for something using that prop.

**Review by andershopkins — APPROVED** [review]

**Pull request merged**